python/cudf/cudf/tests/test_categorical.py

```diff
@@ -875,3 +875,23 @@ def test_cat_groupby_fillna():
         lfunc_args_and_kwargs=(("d",), {}),
         rfunc_args_and_kwargs=(("d",), {}),
     )
+
+
+@pytest.mark.parametrize("op", ["min", "max"])
+def test_categorical_maxima(op):
+    ser = cudf.Series(range(10), dtype="category")
+
+    # NOTE: This test assumes that the default categorical
+    # dtype is unordered. If this behavior changes in
+    # cudf/pandas the test should also change
+    assert not ser.cat.ordered
+
+    # Cannot get maxima of unordered Categorical column
+    with pytest.raises(TypeError, match="Categorical is not ordered"):
+        getattr(ser, op)()
+
+    # Maxima should work after converting to "ordered"
+    ser_pd = ser.to_pandas()
+    result = getattr(ser.cat.as_ordered(), op)()
+    result_pd = getattr(ser_pd.cat.as_ordered(), op)()
```

Comment:
*suggestion*: maybe start with a series where the natural lexical ordering is not the same as that of the categories. e.g.

```
ser = cudf.Series(["a", "d", "c", "z", "g"], dtype=cudf.CategoricalDtype(["z", "c", "g", "d", "a"], ordered=False))
```

That way you also don't need to rely on the default that categoricals are unordered.

Reply:
Ah - Good call. The test fails with this change:

```
>       result = getattr(ser.cat.as_ordered(), op)()

python/cudf/cudf/tests/test_categorical.py:894: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
/datasets/rzamora/miniconda3/envs/cudf_2406/lib/python3.11/site-packages/nvtx/nvtx.py:116: in inner
    result = func(*args, **kwargs)
python/cudf/cudf/core/frame.py:1730: in min
    return self._reduce(
/datasets/rzamora/miniconda3/envs/cudf_2406/lib/python3.11/site-packages/nvtx/nvtx.py:116: in inner
    result = func(*args, **kwargs)
python/cudf/cudf/core/single_column_frame.py:56: in _reduce
    return getattr(self._column, op)(**kwargs)
python/cudf/cudf/core/mixins/mixin_factory.py:11: in wrapper
    return method(self, *args1, *args2, **kwargs1, **kwargs2)
python/cudf/cudf/core/column/categorical.py:721: in _reduce
    return self._encode(
python/cudf/cudf/core/column/categorical.py:884: in _encode
    return self.categories.find_first_value(value)
python/cudf/cudf/core/column/column.py:774: in find_first_value
    first, _ = self._find_first_and_last(value)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <cudf.core.column.string.StringColumn object at 0x7f0e437bf4a0>
[
  "z",
  "c",
  "g",
  "d",
  "a"
]
dtype: object, value = 0

    def _find_first_and_last(self, value: ScalarLike) -> Tuple[int, int]:
        indices = self.indices_of(value)
        if n := len(indices):
            return (
                indices.element_indexing(0),
                indices.element_indexing(n - 1),
            )
        else:
>           raise ValueError(f"Value {value} not found in column")
E           ValueError: Value 0 not found in column

python/cudf/cudf/core/column/column.py:755: ValueError
```

Reply:
Pretty sure we want, `self._encode(...)`? Let me know if I have this wrong.

Reply:
You mean `_decode` (turn an integer code into its value) rather than `_encode` (turn a value into its integer code), but I see that's what you did (I had written `_encode` because of not thinking...)

Reply:
Oops - Yeah, I typed the opposite of what I was thinking :)